рефакторинг кода по результатам урока

alexlarin12 opened the pull request:
1. регистрация ячейки GarageCollectionViewCell без reuseidentifire
2. регистрация ячеек AutoCell и PopoverCell без reuseidentifire
3. обернул констрейнты элементов  в ячейках и контроллерах  в массивы
4. разгрузил от кода ViewDidLoad контроллеров и иниты ячеек
5. небольшой рефакторинг CheckMark
6. исправил баг с неправильным заполнением текстфилдов в AddViewController
7. мелочи: закруглил ячейку GarageCell, исправил орфографию  "case" в "enam"

alexlarin12 committed:
рефакторинг кода по результатам урока

1. регистрация ячейки GarageCollectionViewCell без reuseidentifire
2. регистрация ячеек AutoCell и PopoverCell без reuseidentifire
3. обернул констрейнты элементов  в ячейках и контроллерах  в массивы
4. разгрузил от кода ViewDidLoad контроллеров и иниты ячеек
5. небольшой рефакторинг CheckMark
6. исправил баг с неправильным заполнением текстфилдов в AddViewController
7. мелочи: закруглил ячейку GarageCell, исправил орфографию  "case" в "enam"

aksilont commented on MyCar/App/Flows/Scenes/AddAuto/AddAutoViewController.swift:
```diff
@@ -10,19 +10,23 @@ protocol AddAutoViewControllerDelegate: AnyObject {
-    var indexText: IndexPath?
     var carModel = CarModel(item: "", model: "", number: "", year: "", distance: 0, vin: "", activFlag: false, fuelType: "")
     let carRepository = CarRepository()
     let carParametrsModel = CarParametrsModel()
```
Вот эти модели и сервис по-хорошему вывести во ViewModel, например, AddAutoViewModel и уже с этим работать

aksilont commented on MyCar/App/Flows/Scenes/AddAuto/AddAutoViewController.swift:
```diff
@@ -33,54 +37,53 @@ class AddAutoViewController: UIViewController, UITextFieldDelegate {
-        delegate?.appendAuto(carModel)
+            carRepository.saveCar(carModel: carModel)
+            delegate?.appendAuto(carModel)
             self.navigationController?.popToRootViewController(animated: true)
```
Вот здесь еще надо сделать просто popViewController вместо popToRootViewController

pkslty replied:
Согласен, это путь к перезапуску приложения после сохранения

pkslty commented on MyCar/App/Flows/Scenes/AddAuto/AddAutoViewController.swift:
```diff
@@ -33,54 +37,53 @@ class AddAutoViewController: UIViewController, UITextFieldDelegate {
     @objc func touchReady(){
         if carModel.item != "" && carModel.number != "" {
-        carRepository.saveCar(carModel: carModel)
-        delegate?.appendAuto(carModel)
```
На мой взгляд заморочка с делегатом была лишняя. В гараже можно просто подгружать машины во viewwillappear

pkslty reviewed (changes requested):
Критично:
Поправить возврат по кнопке сохранения
Общие замечания:
1. Поля надо бы проверять на корректность данных (год, например, пробег)
2. При сохранении первого авто ставить ему флаг активного (возможно это должно быть вообще не в этом контроллере)

alexlarin12 committed:
рефакторинг 2

1. переделал переход при нажатии ГОТОВО в AddAuto
2. При добавлении самого первого авто - флаг ставится в true - авто выбрано
3. сделал выбор авто в GarageController

pkslty commented on MyCar/App/Flows/Scenes/Garage/GarageCollectionViewController.swift:
```diff
@@ -91,11 +95,31 @@ class GarageCollectionViewController: UICollectionViewController {
+                cars[i].activFlag = false
+                carRepository.saveCar(carModel: cars[i])
+            }
+            collectionView.reloadData()
```
Как минимум после if - else можно перенести carRepository.saveCar(carModel: cars[i]), чтоб не писать дважды. Но меня все это смущает. Я думаю в репозиторий надо написать метод апдейта модели и апдейтить только две сменившие флаг. Тем более что в дальнейшем захочется редактировать машину, не удалять же снова все по таакому случаю.
